modules/mod-deep-learning/ModelCard.h

```diff
@@ -0,0 +1,212 @@
+
+   void Validate(const Doc& doc, const Doc& schema);
+
+   bool IsLocal() { return m_is_local; }
+   void SetLocal(bool local) { m_is_local = local; }
+
+   std::string GetLocalPath() const { return m_local_path; }
+   void SetLocalPath(const std::string& path) { m_local_path = path; }
+
+public:
+
+   std::string name() const { return m_name; }
+   void name(const std::string &name) { m_name = name; }
+
+   std::string author() const { return m_author; }
+   void author(const std::string &author) { m_author = author; }
+
+   std::string long_description() const { return m_long_description; }
+   void long_description(const std::string &long_description) { m_long_description = long_description; }
+
+   std::string short_description() const { return m_short_description; }
+   void short_description(const std::string &short_description) { m_short_description = short_description; }
+
+   int sample_rate() const { return m_sample_rate; }
+   void sample_rate(int rate) { m_sample_rate = rate; }
+
+   bool multichannel() const { return m_multichannel; }
+   void multichannel(bool multichannel) { m_multichannel = multichannel; }
+
+   std::string effect_type() const { return m_effect_type; }
+   void effect_type(const std::string &type) { m_effect_type = type; }
+
+   const std::vector<std::string> domain_tags() const { return m_domain_tags; }
+   void domain_tags(const std::vector<std::string> tags) { m_domain_tags = tags; }
+
+   const std::vector<std::string> tags() const { return m_tags; }
+   void tags(const std::vector<std::string> tags) { m_tags = tags; }
+
+   const std::vector<std::string> labels() const { return m_labels; }
+   void labels(std::vector<std::string> labels) { m_labels = labels; } // use std::move to not copy
+
+   size_t model_size() const { return m_model_size; }
+   void model_size(size_t model_size) { m_model_size = model_size; }
+
+private:
+   std::string m_name {""};
+   std::string m_author {""};
+   std::string m_long_description {""};
+   std::string m_short_description {""};
+   int m_sample_rate {0};
+   bool m_multichannel  {false};
+   std::string m_effect_type {""};
+   std::vector<std::string> m_domain_tags;
+   std::vector<std::string> m_tags;
+   std::vector<std::string> m_labels;
+   size_t m_model_size {0};
+
+   bool m_is_local {false};
+   std::string m_local_path {""};
+
+public:
+
+   // TODO: add versioning?
+   bool operator==(const ModelCard& that) const;
+   bool operator!=(const ModelCard& that) const;
+};
+
+using ModelCardHolder = std::shared_ptr<ModelCard>;
+using ModelCardFilter = std::function<bool(ModelCardHolder card)>;
+
+class ModelCardCollection final
+{
+public:
+   // set the JSON schema to be used when Validate() is called. 
+   ModelCardCollection() = default;
+
+   // insert a model card into the collection if it matches the schema
+   // will throw if the card does not match the schema
+   // only adds if the card isn't already there (doesn't throw)
+   // @execsafety strong
```

Comment:
Thank you for documenting the exception safety contract!  But here and elsewhere, write it correctly, `@excsafety` not `@execsafety`.  It is supposed to be a promise that the object does not change its state at all in case of exceptions.  It is often implemented by building a copy, and swapping it with the original only when the copy is completed.

But is this comment event right?  I read the implementation and I see no checking of the schema, no throwing of an exception.

That is a Doxygen alias defined in audacity.dox.in.